server/src/main/java/org/opentosca/toscana/plugins/cloudfoundry/client/CloudFoundryInjectionHandler.java

```diff
@@ -0,0 +1,100 @@
+package org.opentosca.toscana.plugins.cloudfoundry.client;
+
+import java.io.FileNotFoundException;
+import java.nio.file.Path;
+import java.nio.file.Paths;
+
+import org.opentosca.toscana.core.plugin.PluginFileAccess;
+import org.opentosca.toscana.plugins.cloudfoundry.application.CloudFoundryApplication;
+import org.opentosca.toscana.plugins.cloudfoundry.application.CloudFoundryService;
+import org.opentosca.toscana.plugins.cloudfoundry.application.CloudFoundryServiceType;
+
+import com.fasterxml.jackson.core.JsonProcessingException;
+import org.json.JSONException;
+import org.slf4j.Logger;
+import org.slf4j.LoggerFactory;
+
+/**
+ inject service credentials to the environment variables
+ depends on the service type
+ */
+public class CloudFoundryInjectionHandler {
+
+    private final static Logger logger = LoggerFactory.getLogger(CloudFoundryInjectionHandler.class);
+    private CloudFoundryApplication app;
+    private CloudFoundryConnection cloudFoundryConnection;
+    private PluginFileAccess fileAccess;
+
+    public CloudFoundryInjectionHandler(PluginFileAccess fileAccess, CloudFoundryApplication app) {
+        this.app = app;
+        this.cloudFoundryConnection = app.getConnection();
+        this.fileAccess = fileAccess;
+    }
+
+    /**
+     create services on CF instance, deploy application, bind application to service and
+     add the credentials to the environment variables.
+     */
+    public void injectServiceCredentials() {
+        deploy();
+        getServiceCredentials();
+    }
+
+    /**
+     deploys the application, creates the services and bind the application to the service
+     the application is not running after that
+     */
+    public Boolean deploy() {
+        Boolean succeed = false;
+        try {
+            if (cloudFoundryConnection != null) {
+                Path pathToApplication = Paths.get(fileAccess.getAbsolutePath(app.getPathToApplication()));
+                succeed = cloudFoundryConnection.pushApplication(pathToApplication,
+                    app.getName(), app.getServicesMatchedToProvider());
+            }
+        } catch (InterruptedException e) {
+            logger.error("Something went wrong while pushing the application");
+            logger.error(e.getMessage());
```

Comment:
Please print the stacktrace of the exception. Just printing the message is like saying "The internet does not work" without giving further information, this makes debugging and understanding Bugs a horrible experience.

A simple solution is replacing the two lines with the following:
```Java
logger.error("Something went wrong while pushing the application", e);
```